### PR TITLE
gui: Add tabs in device editor

### PR DESCRIPTION
**Pull request by qepasa**
### Purpose

Hello everyone. This is long due update to GUI aimed at exposing bandwidth throttling (should have done that long ago, sorry!). I created it following @imsodin suggestion from a few weeks back on IRC. In case you would like to see other things exposed in each tab I'll happily add them.

### Testing

Not much. Clicked through device editor and checked most of its existing functionalities and the new one.

### Screenshots

https://imgur.com/a/EEHGyTY

### CODE REVIEW

**Comment by AudriusButkevicius** [comment]
The screenshots don't look right, errors from wrong tab are visible.

**Comment by AudriusButkevicius** [comment]
Also, I would put compression in advanced.

**Comment by AudriusButkevicius** [comment]
And potentially even addresses.

**Comment by qepasa** [comment]
You're right, I overlooked those errors when I was copying from folder modal. Here is link to screenshots with suggested updates: https://imgur.com/a/hFfJoGG

**Comment by imsodin** [comment]
Maybe add a note on transfer limits about what to enter to not limit at all (is it 0?).

**Comment by AudriusButkevicius** [comment]
also the formatting for the HTML looks wrong, wrong indentation, etc.

**Comment by qepasa** [comment]
Yes, it's 0. I added a note and a tooltip. It looks like this: 
device tab note

I know I screwed up formatting. I foolishly hit reformat file in goland and could not get html formatting settings to indent it properly afterwards.
It looks more or less how it should now although I could have overlooked something.

**Comment by uok** [comment]
A few suggestions - please
- use already existing strings and labels from the general settings modal like "Incoming Rate Limit (KiB/s)" so the translations can be reused.
- keep wording consistent, e.g. use "device rate limits" and not "device bandwidth limits"
- extra text hint about "0" is not needed because
- for red hints just copy texts from general settings "The rate limit must be a non-negative number (0: no limit)"

**Comment by qepasa** [comment]
Thanks for feedback. I added suggested changes. It looks like this now:
gui
gui2

**Comment by imsodin** [comment]
Generally looks good to me, just one (debatable) point. I'd consider compression more "advanced" (as in less used) than address - personally I'd put both into advanced or address into general.

**Comment by uok** [comment]
I think leaving "address" in section general would nudge (new) users to think about setting up direct connections instead of using relay/discovery servers.

**Comment by AudriusButkevicius** [comment]
So this PR itself is ok.

There are plenty of nice things we could expose given we now have the space, but that's a story for a different PR.

As to leaving address in general, people have no clue what dynamic means, and there is already a warning icon if connected not directly in the UI.

File: gui/default/syncthing/device/editDeviceModalView.html [hunk 1]
```diff
@@ -1,98 +1,134 @@
...
+                <div class="col-md-6">
+                  <div ng-class="{'has-error': deviceEditor.maxSendKbps.$invalid && deviceEditor.maxSendKbps.$dirty}">
+                    <div class="row">
+                      <span class="col-md-8" translate>Outgoing Rate Limit (KiB/s)</span>
+                      <div class="col-md-4">
+                        <input name="maxSendKbps" id="maxSendKbps" class="form-control" type="number" ng-model="currentDevice.maxSendKbps" required="" aria-required="true" min="0"/>
+                      </div>
+                    </div>
+                    <p class="help-block" ng-if="!deviceEditor.maxSendKbps.$valid && deviceEditor.maxSendKbps.$dirty" translate>
+                      The rate limit must be a non-negative number (0: no limit) </p>
```
Comment (AudriusButkevicius) [inline gui/default/syncthing/device/editDeviceModalView.html]: The way this is broken over multiple lines makes me sad, please adjust to either `<p>foo</b>` or
```
<p>
   Foo
</p>
```

File: gui/default/syncthing/device/editDeviceModalView.html [hunk 2]
```diff
@@ -127,8 +126,7 @@
                       <input name="maxRecvKbps" id="maxRecvKbps" class="form-control" type="number" ng-model="currentDevice.maxRecvKbps" required="" aria-required="true" min="0"/>
                     </div>
                   </div>
-                  <p class="help-block" ng-if="!deviceEditor.maxRecvKbps.$valid && deviceEditor.maxRecvKbps.$dirty" translate>
-                    The rate limit must be a non-negative number (0: no limit) </p>
+                  <p class="help-block" ng-if="!deviceEditor.maxRecvKbps.$valid && deviceEditor.maxRecvKbps.$dirty" translate>The rate limit must be a non-negative number (0: no limit) </p>
```
Comment (uok) [inline gui/default/syncthing/device/editDeviceModalView.html]: there are spaces before `</p>`- I don't know if this is a problem for translation strings.

File: gui/default/syncthing/device/editDeviceModalView.html [hunk 3]
```diff
@@ -116,8 +116,7 @@
                         <input name="maxSendKbps" id="maxSendKbps" class="form-control" type="number" ng-model="currentDevice.maxSendKbps" required="" aria-required="true" min="0"/>
                       </div>
                     </div>
-                    <p class="help-block" ng-if="!deviceEditor.maxSendKbps.$valid && deviceEditor.maxSendKbps.$dirty" translate>
-                      The rate limit must be a non-negative number (0: no limit) </p>
+                    <p class="help-block" ng-if="!deviceEditor.maxSendKbps.$valid && deviceEditor.maxSendKbps.$dirty" translate>The rate limit must be a non-negative number (0: no limit) </p>
```
Comment (uok) [inline gui/default/syncthing/device/editDeviceModalView.html]: there is a space before `</p>`- I don't know if this is a problem for translation strings.

Comment (AudriusButkevicius) [reply gui/default/syncthing/device/editDeviceModalView.html]: I had the same comment, make sure its identical to the code in the other place.

Comment (qepasa) [reply gui/default/syncthing/device/editDeviceModalView.html]: There are a few such spaces before closing tags in current device modal but not in folder modal. I will get rid of all just to be safe.

Comment (qepasa) [reply gui/default/syncthing/device/editDeviceModalView.html]: Well to be exact there are no such spaces in code from master. I must have added them accidentally when I was fooling around with formatting. Anyway, they are gone.

**Comment by AudriusButkevicius** [comment]
This seems to affect one other string (middle) I did not expect:

image

It was double space at one place before, now it's removed, but that means everyone will have to retranslate, so we should change it back.

**Comment by AudriusButkevicius** [comment]
Jesus fucking christ, the CI builds are now taking 20 minutes. Is the code base being transpiled to C++ before it gets compiled or something?

**Pull request merged**